tests/baselines/reference/unionThisTypeInFunctions.errors.txt

```diff
@@ -0,0 +1,34 @@
+tests/cases/conformance/types/thisType/unionThisTypeInFunctions.ts(10,5): error TS2684: The 'this' context of type 'Real | Fake' is not assignable to method's 'this' of type 'Real & Fake'.
+  Type 'Real' is not assignable to type 'Real & Fake'.
+    Type 'Real' is not assignable to type 'Fake'.
+      Types of property 'method' are incompatible.
+        Type '(this: Real, n: number) => void' is not assignable to type '(this: Fake, n: number) => void'.
+          The 'this' types of each signature are incompatible.
+            Type 'Fake' is not assignable to type 'Real'.
+              Types of property 'data' are incompatible.
+                Type 'number' is not assignable to type 'string'.
+
+
+==== tests/cases/conformance/types/thisType/unionThisTypeInFunctions.ts (1 errors) ====
+    interface Real {
+        method(this: this, n: number): void;
+        data: string;
+    }
+    interface Fake {
+        method(this: this, n: number): void;
+        data: number;
+    }
+    function test(r: Real | Fake) {
+        r.method(12); // error

```

Comment:
Not a big fan of this, since it's prima facie Obviously Legal.

Reply:
I mean, this is the crux of the change 😅

Reply:
Well, it's Obviously Legal and Technically Incorrect, and in this case I think Technically Incorrect should win.